columnflow/tasks/reduction.py

```diff
@@ -198,18 +199,7 @@ def run(self):
             # loop through all object selection, go through their masks
             # and create new collections if required
             if "objects" in sel.fields:
-                for src_name in sel.objects.fields:
-                    # get all destination collections, handling those named identically to the
-                    # source collection last
-                    dst_names = list(sel["objects", src_name].fields)
-                    if src_name in dst_names:
-                        # move to the end
-                        dst_names.remove(src_name)
-                        dst_names.append(src_name)
-                    for dst_name in dst_names:
-                        object_mask = sel.objects[src_name, dst_name][event_mask]
-                        dst_collection = events[src_name][object_mask]
-                        events = set_ak_column(events, dst_name, dst_collection)
+                events = create_collections_from_masks(events, sel.objects, event_mask=event_mask)
```

Comment:
Related to the above comment on `event_mask`, it should be possible to pass `sel.objects[event_mask]` here instead of handling the `event_mask` internally in `create_collections_from_mask`.

Reply:
Yup. The comment I made above applies here as well.